modules/fb_instant_articles_views/views/handler/views_handler_filter_enabled_fia.inc

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/**
+ * @file
+ * Definition of views_handler_filter_enabled_fia.
+ */
+
+
+/**
+ * Simple filter to select entities marked as beeing published for fbia.
+ */
+class views_handler_filter_enabled_fia extends views_handler_filter {
+
+
+  /**
+   * Display the filter on the administrative summary
+   */
+  function admin_summary() {
+    return t('Filtering for selected Facebook Instant Articles content.');
+  }
+
+  /**
+   * Determine if a filter can be exposed.
+   */
+  function can_expose() {
+    return FALSE;
+  }
+
+  /**
+   * Determine if a filter can be converted into a group.
+   * Only exposed filters with operators available can be converted into groups.
+   */
+  function can_build_group() {
+    return FALSE;
+  }
+
+  /**
+   * Provide the basic form which calls through to subforms.
+   * If overridden, it is best to call through to the parent,
+   * or to at least make sure all of the functions in this form
+   * are called.
+   */
+  function options_form(&$form, &$form_state) {
+    parent::options_form($form, $form_state);
+    $form['info'] = array(
+      '#type' => 'markup',
+      '#markup' => 'Filter for selected content shown in Facebook Instant Articles feed. Filter only works if "All nodes enabled by default?" is unchecked.',
+      '#prefix' => '<div class="clearfix">',
+      '#suffix' => '</div>',
+    );
+  }
+
+  /**
+   * Add this filter to the query.
+   */
+  function query() {
+    // We don't extend the query if the setting for all_enabled_default is set.
+    $default_enabled = variable_get('fb_instant_articles_all_enabled_default', TRUE);
```

Comment:
Hum, this variable no longer exists after the refactor.  It's new name is `fb_instant_articles_rss_all_enabled_default` and it's a part of the `fb_instant_articles_rss` module.  The new thinking in the refactor was the `fb_instant_articles_rss` implements a dead simple RSS feed that internalizes everything it needs for the rudimentary filtering it does, and in the views module, site builders, if they needed it, would define whatever filtering they need.  For instance on the site I'm working on, I've added a boolean field to our blog nodes for "Publish to Facebook Instant Articles", then I filter the view based on that.

As such, I don't think that this filter is necessary anymore.  At least, it doesn't make sense without the `fb_instant_articles_rss` enabled, and even then, the presumption is that you either use `fb_instant_articles_rss` or `fb_instant_articles_views` and not both :confused:.

Does that make sense?